src/Documenter.jl

```diff
@@ -500,6 +457,20 @@ function deploydocs(;
         end
         # Change to the root directory and try to deploy the docs.
         cd(root) do
+            # Find the commit sha.
+            # We'll make sure we run the git commands in the source directory (root), in case
+            # the working directory has been changed (e.g. if the makedocs' build argument is
+            # outside root).
+            sha = try
+                readchomp(`git rev-parse --short HEAD`)
+            catch
+                # git rev-parse will throw an error and return code 128 if it is not being
+                # run in a git repository, which will make run/readchomp throw an exception.
+                # We'll assume that if readchomp fails it is due to this and set the sha
+                # variable accordingly.
+                "(not-git-repo)"
+            end
```

Comment:
This conflicts with master?

Reply:
Yea, this PR is mostly a playground, review #1147 instead :p